crates/web5/src/lib.rs

```diff
@@ -4,8 +4,25 @@ pub mod dids;
 
 mod datetime;
 pub mod errors;
+mod http;
 pub mod jose;
 pub mod json;
 
+pub use http::set_http_client;
```

Comment:
don't we also need to expose the `HttpClient` trait in order for developers to bring their own?

IMO we should simply `pub mod http;` everything, otherwise in `tbdex-rs` we're going to have to copy/pasta all the http code for it's own BYO functionality (which over there is where the hard requirement kicks in because we have to implement a foreign fetch over the wasm ffi)